examples/listbox_sort_stringlist/main.rs

```diff
@@ -0,0 +1,147 @@
+        .title("Sorted StringList")
+        .build();
+
+    let vbox = gtk::Box::new(gtk::Orientation::Vertical, 5);
+
+    // Create a StringSorter with a property expression to sort
+    // StringObjects in a StringList.  StringObject has a "string" property.
+    let expression = gtk::PropertyExpression::new(
+        gtk::StringObject::static_type(),
+        None::<gtk::Expression>,
+        "string",
+    );
+    let sorter = gtk::StringSorter::new(Some(expression));
+    sorter.set_ignore_case(true);
+
+    // Create our list store as a StringList and populate with some strings.
+    let model = gtk::StringList::new(&["zoo", "abba", "donkey", "sunrise", "river", "phoenix"]);
+
+    // Create a sort model and bind it to the ListStore and the sorter.
+    let sort_model = gtk::SortListModel::new(Some(model.clone()), Some(sorter.clone()));
+
+    // And then create the UI part, the listbox and bind the sort
+    // model to it. Whenever the UI needs to show a new row, e.g. because
+    // it was notified that the model changed, it will call the callback
+    // with the corresponding item from the model and will ask for a new
+    // gtk::ListBoxRow that should be displayed.
+    //
+    // The gtk::ListBoxRow can contain any possible widgets.
+    //  Here we use an Inscription.
+
+    let listbox = gtk::ListBox::new();
+    listbox.bind_model(
+        Some(&sort_model),
+        clone!(
+            #[weak(rename_to = _panel)]
+            window,
+            #[upgrade_or_panic]
+            move |obj| {
+                let list_object = obj
+                    .downcast_ref::<gtk::StringObject>()
+                    .expect("The object should be of type `StringObject`.");
+                let row = gtk::Inscription::new(Some(&list_object.string()));
+                row.set_xalign(0.0);
+                row.upcast()
+            }
+        ),
+    );
+
+    let scrolled_window = gtk::ScrolledWindow::builder()
+        .hscrollbar_policy(gtk::PolicyType::Never) // Disable horizontal scrolling
+        .min_content_height(480)
+        .min_content_width(360)
+        .build();
+
+    scrolled_window.set_child(Some(&listbox));
+
+    let hbox = gtk::Box::new(gtk::Orientation::Horizontal, 5);
+
+    // The add button opens a new dialog which is basically the same as the edit
+    // dialog, except that we don't have a corresponding item yet at that point
+    // and only create it once the Ok button in the dialog is clicked, and only
+    // then add it to the model. Once added to the model, it will immediately
+    // appear in the listbox UI
+    let add_button = gtk::Button::with_label("Add");
+    add_button.set_sensitive(false);
+
+    hbox.append(&add_button);
+
+    // Via the delete button we delete the item from the model that
+    // is at the index of the selected row. Also deleting from the
+    // model is immediately reflected in the listbox.
+    let delete_button = gtk::Button::with_label("Delete");
```

Comment:
I feel like this adds more complexity and it is not really the best way to achieve that. So I would just keep it for implementing a sort with stringlist

Reply:
Thanks for your suggestions.  I've removed the "add" and "delete" buttons to simplify the example.